### PR TITLE
fix: Add mode property to forkApplicationSaga

### LINKED ISSUE #35835: [Bug]: `400 - Bad request` error seen while forking apps from within the application
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we try to fork an app from within the application, we see a 400 - Bad request error. The same error is not seen when we try to fork the same app from the homepage.

### Steps To Reproduce

1. Go to an app & click on the app name dropdown.
2. Click on Fork application & select a workspace.

**Expected:** The application should be forked without an errors.
**Actual:** We see a 400 - Bad request error.

### Public Sample App

_No response_

### Environment

Production

### Severity

Critical (Broken Production apps)

### Issue video log

https://jam.dev/c/c55c5425-7bbc-49ab-a6fb-e3d63b21befd

### Version

Cloud

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This pull request adds the `mode` property to the `forkApplicationSaga` function in the `ApplicationSagas.tsx` file. 

The `mode` property is set to `APP_MODE.EDIT`, allowing for better control and customization of the application editing mode.

Fixes #35835
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Fork"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10507731953>
> Commit: eba9176ad4d7696bd8b1095bd302831084dc5cce
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10507731953&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Fork`
> Spec:
> <hr>Thu, 22 Aug 2024 12:33:03 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced the application forking functionality by introducing an edit mode, allowing for more flexible handling of different operational states during the forking process.
	- Improved state management in the page list by adding properties for tracking the current page and base page ID, enhancing user interaction with the page list.

### CODE REVIEW

**Review by jacquesikot — APPROVED** [review]

**Comment by jacquesikot** [comment]
@rahulbarwal Why is this change just added? There is an if check for this `action.payload.editMode` in the code, so why did we not need to pass this before, but now we need to pass it, what changed? Or did we just miss it earlier?

**Comment by rahulbarwal** [comment]
> @rahulbarwal Why is this change just added? There is an if check for this `action.payload.editMode` in the code, so why did we not need to pass this before, but now we need to pass it, what changed? Or did we just miss it earlier?

@jacquesikot I ran a git bisect to find the answer to this very question. It points to default resources PR: https://github.com/appsmithorg/appsmith/pull/34765
My guess is that prior to this the PagesAPI did not care if you missed `mode` parameter and now it does and errors out.
@nidhi-nair to confirm.

File: app/client/src/reducers/entityReducers/pageListReducer.tsx [hunk 1]
```diff
@@ -61,6 +61,8 @@ export const pageListReducer = createReducer(initialState, {
       ...action.payload,
       defaultPageId: defaultPage?.pageId,
       defaultBasePageId: defaultPage?.basePageId,
+      currentBasePageId: defaultPage?.basePageId,
```
Comment (nidhi-nair) [inline app/client/src/reducers/entityReducers/pageListReducer.tsx:64]: What do these properties help with @rahulbarwal ?

Comment (rahulbarwal) [reply app/client/src/reducers/entityReducers/pageListReducer.tsx:64]: @nidhi-nair 

These properties are used by editor header to show the current page at top.
Basically, after adding edit mode, the page name in the list was not updating correctly, and reason was that it used `currentPageId` property.
When we forked an app form editor, it already has `currentPageId` which was not getting updated.

Comment (nidhi-nair) [reply app/client/src/reducers/entityReducers/pageListReducer.tsx:64]: Gotcha, thanks! I'm approving the PR as a quick fix, but is there a way to add tests to these flows as well?

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**